src/promptflow/promptflow/_sdk/_service/utils/utils.py

```diff
@@ -105,13 +108,23 @@ def get_random_port():
 
 
 def _get_process_by_port(port):
-    for proc in psutil.process_iter(["pid", "connections", "create_time"]):
-        try:
-            for connection in proc.connections():
-                if connection.laddr.port == port:
-                    return proc
-        except psutil.AccessDenied:
-            pass
+    # use net_connections api to accelerate the process, but require root privileges on macOS and AIX. So use original
+    # way to get process on these platforms. Note: (Solaris) UNIX sockets are not supported, (OpenBSD) laddr and raddr
+    # fields for UNIX sockets are always set to “”. This is a limitation of the OS.
+    # Refer here for more details: https://psutil.readthedocs.io/en/latest/#psutil.AccessDenied
+    try:
+        for conn in psutil.net_connections():
```

Comment:
shall we do if checking on the os platform, instead of excetp here?

Reply:
offline synced, keep this behavior for now